src/gf.c

```diff
@@ -2650,13 +2652,15 @@ jl_code_instance_t *jl_compile_method_internal(jl_method_instance_t *mi, size_t
         }
 
         JL_GC_PUSH1(&codeinst);
+        double compile_time = jl_hrtime();
         int did_compile = jl_compile_codeinst(codeinst);
+        compile_time = jl_hrtime() - compile_time;
```

Comment:
@IanButterworth: Can you help me understand why we have so many different variants of measuring this metric (compilation time)?

I was surprised to see we had to add a new start/stop measurement here. We already have a global metric for compilation time, which has timings around inference and llvm codegen. Why didn't we add to the per-method-instance compilation time counters from those places?

Or, conversely, if this spot can correctly capture all compilation time for a given method instance, can we delete those other spots and incremental global compilation time here?

- https://github.com/JuliaLang/julia/blob/fe864374d50b121aeb3ce5473c206338b21df5e5/src/gf.c#L4390-L4396
- https://github.com/JuliaLang/julia/blob/a98f3713cb9aea3225cdb0e3ec0dd719f556a974/src/aotcompile.cpp#L571-L572
- https://github.com/JuliaLang/julia/blob/a98f3713cb9aea3225cdb0e3ec0dd719f556a974/src/jitlayers.cpp#L578-L579
- ... etc

What's the difference between these approaches? Can we unify them?
Thanks!

Reply:
Maybe that can be reused, yeah. The timer added here seems lower cost than atomic fetches etc. but perhaps it's overcounting.